connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/ProtobufDeserializer.scala

```diff
@@ -193,6 +193,11 @@ private[sql] class ProtobufDeserializer(
       case (INT, ShortType) =>
         (updater, ordinal, value) => updater.setShort(ordinal, value.asInstanceOf[Short])
 
+      case (INT, LongType) =>
+        (updater, ordinal, value) =>
+          updater.setLong(
+            ordinal,
+            Integer.toUnsignedLong(value.asInstanceOf[Int]))
```

Comment:
Hm, I wouldn't add this. It would be problematic when Spark has unsigned types. For the same reason, Parquet also doesn't support unsigned physical types for Spark.

Reply:
> It would be problematic when Spark has unsigned types. For the same reason, Parquet also doesn't support unsigned physical types for Spark.

hey, i'm not sure if i follow; do you mind explaining what you mean by this?

My goal here is to add an option allowing unsigned 32 and 64 bit integers coming from protobuf to be represented in a type that can contain them without overflow. I actually modeled my code off of how the parquet code today is written, which i believe is doing this same thing by default:

https://github.com/justaparth/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaConverter.scala#L243-L270

https://github.com/justaparth/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRowConverter.scala#L345-L351

Reply:
Oh, it was added at here https://github.com/apache/spark/pull/31921. My memory was old.

Reply:
okay, cc @yaooqinn @cloud-fan

Reply:
I am less certain about over all Spark context, but for `from_protobuf()` this looks fine so far (since it is only enabled with an option).